src/main/java/taxi/service/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package taxi.service;
+
+import java.util.Optional;
+import taxi.exception.AuthenticationException;
+import taxi.lib.Inject;
+import taxi.lib.Service;
+import taxi.model.Driver;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverService driverService;
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Optional<Driver> driver = driverService.findDriverByLogin(login);
+        if (driver.get().getPassword().equals(password)) {
```

Comment:
You may have a situation in which the optional will be empty, take this into account in your test

Reply:
Done